assessment/report_generator.go

```diff
@@ -189,3 +196,106 @@ func sourceColumnDefinitionToString(columnDefinition utils.SrcColumnDetails) str
 	}
 	return s
 }
+
+// TODO move calculation logic to assessment engine
+func calculateTableDbChangesAndImpact(srcTable utils.TableDetails, spTable utils.TableDetails) (string, string) {
+	changes := []string{}
+	impact := []string{}
+	if !strings.Contains(srcTable.Charset, "utf8") { // TODO add charset level comparisons - per source
+		changes = append(changes, "charset")
+		impact = append(impact, "storage increase")
+	}
+	if len(changes) == 0 {
+		changes = append(changes, "None")
+	}
+	return strings.Join(changes, ","), strings.Join(impact, ",")
+}
+
+// TODO move calculation logic to assessment engine
+func calculateColumnDbChangesAndImpact(srcCol utils.SrcColumnDetails, spCol utils.SpColumnDetails) (string, string) {
+	changes := []string{}
+	impact := []string{}
+	if srcCol.Datatype != spCol.Datatype { // TODO type specific checks on size
+		changes = append(changes, "type")
+
+		if getSrcColSizeBytesMySQL(srcCol) < getSpColSizeBytes(spCol) {
+			impact = append(impact, "storage increase")
+		}
+	}
+
+	//TODO Add check for unsigned
+	//TODO add check for not null to null scenarios
+	//TODO add check for size overflow
+	//TODO add diffs in modifiers and features - generated cols, auto inc, default etc
+
+	if len(changes) == 0 {
+		changes = append(changes, "None")
+	}
+	if len(impact) == 0 {
+		impact = append(impact, "None")
+	}
+	return strings.Join(changes, ","), strings.Join(impact, ",")
+}
+
+// TODO - move to source specific interfaces - also account for charsets
+func getSrcColSizeBytesMySQL(srcCol utils.SrcColumnDetails) int64 {
+	switch strings.ToLower(srcCol.Datatype) {
+	case "date":
+		return 4
+	case "timestamp":
+		return 4
+	case "bit":
```

Comment:
Following [this](https://www.mysqldatatypes.com/), maybe we should have : (srcCol.Mods[0] + 7)/ 8

Reply:
changed.